### PR TITLE
update area of specialisation of doctor 

### LINKED ISSUE #7612: Areas of Specializations needs to Be increased to a Doctor w.r.t. facility Details
When Updating the facility details, while adding the Doctors list currently it is observed only few areas of Specialization are found in dropdowns.
This has to be incremented with various Specialization skills. Attaching the respective Snapshot for reference
Screenshot 2024-04-15 114410

### PR DESCRIPTION
**Pull request by rash-27**
## Proposed Changes

- Fixes #7612 
- Linked backend pr https://github.com/coronasafe/care/pull/2102
- [x] Facility dashboard (https://github.com/coronasafe/care_fe/issues/7616) change "Doctor List" to "Staff Capacity"
- [x] Change the text on the button for "Add doctor type" to "Add Staff type"
- [x] edit the pop-up window. title to be "Add staff capacity"
- [x] "area of specialisation" to be changed to "Staff type"
- [x] the dropdown for area of specialisation to be synced with the list of skills available on user module + Others+ Nurse.
- [x] the existing data to be mapped as "Critical care"= "Intensivist", "pediatric"="Pediatrician". "other speciality"="Others"

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by rash-27** [comment]
@nihal467 @aparnacoronasafe In this issue, we are synchronising the doctor types with available skills and using staff capacity instead of the doctor's list in the facility dashboard. I feel like we can completely go with the staff capacity in the facility dashboard and make corresponding changes to shift from displaying doctors to staff.The changes include 
- [ ] Changing the route names that we are using currently using to fetch staff details (`/api/v1/facility/{facilityId}/hospital_doctor/` and `/api/v1/facility/{facilityId}/hospital_doctor/{id}/`) for more readability 
- [ ] And also making changes in the frontend like, the notifications currently are displaying "Doctor count added successfully" and we can change them to "Staff count added successfully" and etc. 
- [ ] And also in frontend code can we change the variable names accordingly for more readability of the code as currently we are displaying the Staff details in facility dashboard instead of doctors details ?
 What do you think?

**Comment by nihal467** [comment]
@rash-27 any update on this PR

**Comment by nihal467** [comment]
@rithviknishad can you update the following bugs

**Comment by nihal467** [comment]
image

- [x] change all doctor related text to staff related text in the facility create work flow

**Comment by nihal467** [comment]
image

- [ ] modify the delete pop-up text and relate notification (create, update, and delete)

**Comment by nihal467** [comment]
image

- [x] update the existing data modify pop-up and update doctor capacity capacity

**Comment by nihal467** [comment]
image

- [x] change it to total staff count

File: src/Common/constants.tsx [hunk 1]
```diff
@@ -230,11 +230,62 @@ export const getBedTypes = ({
 };
 
 export const DOCTOR_SPECIALIZATION: Array<OptionsType> = [
-  { id: 1, text: "General Medicine", desc: "bg-doctors-general" },
-  { id: 2, text: "Pulmonology", desc: "bg-doctors-pulmonology" },
-  { id: 3, text: "Critical Care", desc: "bg-doctors-critical" },
-  { id: 4, text: "Paediatrics", desc: "bg-doctors-paediatrics" },
-  { id: 5, text: "Other Speciality", desc: "bg-doctors-other" },
+  { id: 1, text: "General Medicine Doctors", desc: "bg-doctors-general" },
```
Comment (nihal467) [inline src/Common/constants.tsx]:
```suggestion
  { id: 1, text: "General Medicine", desc: "bg-doctors-general" },
```

File: src/Common/constants.tsx [hunk 2]
```diff
@@ -230,11 +230,62 @@ export const getBedTypes = ({
...
+  { id: 2, text: "Pulmonology Doctors", desc: "bg-doctors-general" },
```
Comment (nihal467) [inline src/Common/constants.tsx]:
```suggestion
  { id: 2, text: "Pulmonology", desc: "bg-doctors-general" },
```

**Review by nihal467 — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Review by nihal467 — APPROVED** [review]

**Pull request merged**